app/services/github_client/user.rb

```diff
@@ -4,11 +4,23 @@ def initialize(username)
       @username = username
     end
 
-    def repositories
-      response = connection.get("/users/#{@username}/repos") do |request|
-        request.params['type'] = 'member'
+    def pull_request_events
+      events = (1..10).map do |page|
+        response = connection.get("/users/#{@username}/events/public") do |request|
+          request.params['page'] = page
+        end
+        results = JSON.parse(response.body, symbolize_names: true)
+
+        if results.empty? || older_than_a_day(results.last[:created_at]) || results.size < 30
```

Comment:
you added `older_than_a_day` condition because there will be a scheduled task running once per day? What if for some reason it fails one day and doesn't run? What other options do we have? Maybe storing the date of the last run?

Reply:
I'm not sure I like the idea of tying this method to that time restriction. If we wanted to do a backfill we'd have to get rid of that, for example. I'd like it to be optional.
But on a second thought, I think we should be fine if we wanted to add that in another PR down the line.

Reply:
I guess storing the date of the last run may be the best choice. I added this because iterating over the 10 pages every time we run this, takes a lot of time. We have like 80 users and it would imply doing around 800 requests. We may even have some issues with API requests per minute restrictions